packages/cli/src/execution-lifecycle-hooks/shared/__tests__/shared-hook-functions.test.ts

```diff
@@ -0,0 +1,37 @@
+import { mock } from 'jest-mock-extended';
+import type { IRun } from 'n8n-workflow';
+import { NodeOperationError } from 'n8n-workflow';
+
+import { determineFinalExecutionStatus } from '../shared-hook-functions';
+
+describe('determineFinalExecutionStatus', () => {
+	describe('When waitTill is not set', () => {
+		test.each(['canceled', 'crashed', 'error', 'success'])('should return "%s"', (status) => {
+			const runData = { status, data: {} } as IRun;
+			expect(determineFinalExecutionStatus(runData)).toBe(status);
+		});
+	});
+
+	it('should return "error" when resultData.error exists', () => {
+		const runData = {
+			status: 'running',
+			data: {
+				resultData: {
+					error: new NodeOperationError(mock(), 'An error occurred'),
+				},
+			},
+		} as IRun;
```

Comment:
Nit - Let's use `mock<IRun>({ ... })` instead of asserting?

Reply:
using `mock` in this case required setting quite a few properties explicitly as `undefined` because `mock` returns recursive proxy for all non-defined properties.

I'll try to clean this up in a future PR.